source/math/FloatPoint3D.ooc

```diff
@@ -35,12 +35,17 @@ FloatPoint3D: cover {
 	init: func@ (=x, =y, =z)
 	init: func@ ~default { this init(0.0f, 0.0f, 0.0f) }
 	init: func@ ~fromPoint2D (point: FloatPoint2D, z := 0.0f) { this init(point x, point y, z) }
+	pNorm: func (p: Float) -> Float {
+		p == 1 ?
+		this x abs() + this y abs() + this z abs() :
+		(this x abs() pow(p) + this y abs() pow(p) + this z abs() pow(p)) pow(1.0f / p)
```

Comment:
Putting everything on the same line is shorter than line 42, but maybe it isn't clear enough.